main.go

```diff
@@ -175,6 +172,67 @@ func main() {
 	}
 }
 
+func prepareTraceMode(traceString string) (uint32, []int, error) {
+	// Set Default mode - all new Processes only
+	mode := tracee.ModeProcessNew
+	var pidsToTrace []int
+	traceHelp := "--trace can be the following options:\n"
+	traceHelp += "<blank> or 'p' or 'process' or 'process:new' | Trace new processes\n"
+	traceHelp += "'process:all'                                | Trace all processes\n"
+	traceHelp += "'process:<pid>,<pid2>,...' or 'p:<pid>,...'  | Trace specific PIDs\n"
+	traceHelp += "'c' or 'container' or 'container:new'        | Trace new containers\n"
+	traceHelp += "'container:all'                              | Trace all containers\n"
+	if traceString == "help" {
+		return 0, nil, fmt.Errorf(traceHelp)
+	}
+
+	traceSplit := strings.Split(traceString, ":")
+
+	// Get The trace type - process or  container
+	traceType := traceSplit[0]
+	if traceType != "process" && traceType != "container" && traceType != "p" && traceType != "c" {
+		return 0, nil, fmt.Errorf(traceHelp)
+	}
+	traceType = string(traceType[0])
+
+	// Get The trace option, default is 'new' for all trace types:
+	traceOption := "new"
+	if len(traceSplit) == 2 {
+		traceOption = traceSplit[1]
+	} else if len(traceSplit) > 2 {
+		return 0, nil, fmt.Errorf(traceHelp)
+	}
+
+	// Convert to Traceing Mode
+	if traceType == "p" {
+		if traceOption == "all" {
+			mode = tracee.ModeProcessAll
+		} else if traceOption == "new" {
+			mode = tracee.ModeProcessNew
+		} else {
+			mode = tracee.ModeProcessList
+			// Attempt to split into PIDs
+			for _, pidString := range strings.Split(traceOption, ",") {
```

Comment:
What about the case that someone gives "process:"?
In that case, traceOption is an empty string - and tracee will trace nothing

Reply:
Good catch, will fix